adapters/YahooAdapter/YahooAdapter/Native/GADMAdapterYahooNativeAd.m

```diff
@@ -0,0 +1,345 @@
+}
+
+- (GADNativeAdImage *)icon {
+  return [self imageForComponent:@"iconImage"];
+}
+
+- (nullable UIView *)mediaView {
+  id<YASComponent> component = [_nativeAd component:@"video"];
+  if (component) {
+    id<YASNativeViewComponent> videoComponent = (id<YASNativeViewComponent>)component;
+    if ([videoComponent conformsToProtocol:@protocol(YASNativeVideoComponent)]) {
+      _yahooVideoPlayerView = [[YASVideoPlayerView alloc] init];
+      YASErrorInfo *error =
+          [(id<YASNativeVideoComponent>)videoComponent prepareView:_yahooVideoPlayerView];
+      if (error) {
+        NSLog(@"[YahooAdapter]: Failed to render mediaView video: %@", error);
+      }
+      return _yahooVideoPlayerView;
+    }
+  }
+
+  component = [_nativeAd component:@"mainImage"];
+  if (component) {
+    id<YASNativeViewComponent> mainImageComponent = (id<YASNativeViewComponent>)component;
+    if ([mainImageComponent conformsToProtocol:@protocol(YASNativeImageComponent)]) {
+      UIImageView *mainImageView = [[UIImageView alloc] init];
+      YASErrorInfo *error =
+          [(id<YASNativeImageComponent>)mainImageComponent prepareView:mainImageView];
+      mainImageView.userInteractionEnabled = NO;
+      if (error) {
+        NSLog(@"[YahooAdapter]: Failed to render mediaView image: %@", error);
+      }
+      return mainImageView;
+    }
+  }
+
+  return nil;
+}
+
+- (CGFloat)mediaContentAspectRatio {
+  id<YASComponent> component = [_nativeAd component:@"video"];
+  if ([component conformsToProtocol:@protocol(YASNativeVideoComponent)]) {
+    id<YASNativeVideoComponent> videoComponent = (id<YASNativeVideoComponent>)component;
+    if (videoComponent.height != 0.0f) {
+      return (videoComponent.width / videoComponent.height);
+    }
+  }
+  return 0.0f;
+}
+
+- (BOOL)hasVideoContent {
+  return self.mediaView != nil;
+}
+
+- (NSDecimalNumber *)starRating {
+  NSString *ratingString = [self stringForComponent:@"rating"];
+  if (ratingString.length > 0) {
+    NSInteger stars = 0;
+    NSInteger total = 0;
+    NSScanner *scanner = [NSScanner scannerWithString:ratingString];
+
+    NSMutableCharacterSet *set = [[NSMutableCharacterSet alloc] init];
+    [set formUnionWithCharacterSet:[NSCharacterSet letterCharacterSet]];
+    [set formUnionWithCharacterSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+    [scanner setCharactersToBeSkipped:set];
+
+    if ([scanner scanInteger:&stars] && [scanner scanInteger:&total]) {
+      return [NSDecimalNumber
+          decimalNumberWithString:[NSString stringWithFormat:@"%ld.%ld", (long)stars, (long)total]];
+    }
+  }
+  return nil;
+}
+
+- (nullable NSDictionary *)extraAssets {
+  return nil;
+}
+
+- (void)didRecordImpression {
+  [_nativeAd fireImpression];
+}
+
+- (void)didRecordClickOnAssetWithName:(GADNativeAssetIdentifier)assetName
+                                 view:(UIView *)view
+                       viewController:(UIViewController *)viewController {
+  [_nativeAd invokeDefaultAction];
```

Comment:
I do not see this method being called when the ad or its components are clicked.  

So, a question: I have a draft adapter working with Google Mobile Ads SDK v8.9.0, where I do see this method being called, though 'handleUserClicks' is not implemented.  In this PR's adapter using Google Ads v10.0.0,  if I add 'handleUserClicks' and return NO, I still don't see this method being called.   

There are other ways to set up the native ad text labels and image views if we expect the Yahoo SDK to handle user clicks, please see https://sdk.yahooinc.com/yahoo-ads/native.html and the Yahoo Sample App https://github.com/yahoo/YahooMobileSDK-SampleApps-iOS/blob/main/ObjC/SampleApp/NativeContainerView.m, in particular the 'prepareView', 'prepareLabel', and 'prepareButton' APIs.

However, if we don't intend to have Yahoo handle clicks, I would expect this method to be called, perhaps I'm overlooking something.

Reply:
This should now be called for all assets.